pkg/internal/apis/certmanager/validation/certificaterequest_test.go

```diff
@@ -100,136 +105,651 @@ func TestValidateCertificateRequestUpdate(t *testing.T) {
+				Spec: cminternal.CertificateRequestSpec{
+					Request:   mustGenerateCSR(t, gen.Certificate("spec", gen.SetCertificateDNSNames("example.com"), gen.SetCertificateKeyUsages(cmapi.UsageAny), gen.SetCertificateIsCA(true))),
+					IssuerRef: validIssuerRef,
+					IsCA:      true,
+					Usages:    []cminternal.KeyUsage{cminternal.UsageAny},
+				},
+				Status: cminternal.CertificateRequestStatus{
+					Conditions: []cminternal.CertificateRequestCondition{
+						{
+							Type:   cminternal.CertificateRequestConditionApproved,
+							Status: cminternalmeta.ConditionTrue,
+							Reason: "cert-manager.io",
+						},
+						{
+							Type:   cminternal.CertificateRequestConditionDenied,
+							Status: cminternalmeta.ConditionTrue,
+							Reason: "Foo",
+						},
+					},
+				},
+			},
+			want: []*field.Error{
+				field.Forbidden(fldPathConditions, "both 'Denied' and 'Approved' conditions cannot coexist"),
+			},
+		},
+		"CertificateRequest with multiple Approved conditions, should error": {
+			cr: &cminternal.CertificateRequest{
+				Spec: cminternal.CertificateRequestSpec{
+					Request:   mustGenerateCSR(t, gen.Certificate("spec", gen.SetCertificateDNSNames("example.com"), gen.SetCertificateKeyUsages(cmapi.UsageAny), gen.SetCertificateIsCA(true))),
+					IssuerRef: validIssuerRef,
+					IsCA:      true,
+					Usages:    []cminternal.KeyUsage{cminternal.UsageAny},
+				},
+				Status: cminternal.CertificateRequestStatus{
+					Conditions: []cminternal.CertificateRequestCondition{
+						{
+							Type:   cminternal.CertificateRequestConditionApproved,
+							Status: cminternalmeta.ConditionTrue,
+							Reason: "cert-manager.io",
+						},
+						{
+							Type:   cminternal.CertificateRequestConditionApproved,
+							Status: cminternalmeta.ConditionFalse,
+							Reason: "foo",
+						},
+					},
+				},
+			},
+			want: []*field.Error{
+				field.Forbidden(fldPathConditions, `multiple "Approved" conditions present`),
+			},
+		},
+		"CertificateRequest with multiple Denied conditions, should error": {
+			cr: &cminternal.CertificateRequest{
+				Spec: cminternal.CertificateRequestSpec{
+					Request:   mustGenerateCSR(t, gen.Certificate("spec", gen.SetCertificateDNSNames("example.com"), gen.SetCertificateKeyUsages(cmapi.UsageAny), gen.SetCertificateIsCA(true))),
+					IssuerRef: validIssuerRef,
+					IsCA:      true,
+					Usages:    []cminternal.KeyUsage{cminternal.UsageAny},
+				},
+				Status: cminternal.CertificateRequestStatus{
+					Conditions: []cminternal.CertificateRequestCondition{
+						{
+							Type:   cminternal.CertificateRequestConditionDenied,
+							Status: cminternalmeta.ConditionTrue,
+							Reason: "Foo",
+						},
+						{
+							Type:   cminternal.CertificateRequestConditionDenied,
+							Status: cminternalmeta.ConditionFalse,
+							Reason: "Foo",
+						},
+					},
+				},
+			},
+			want: []*field.Error{
+				field.Forbidden(fldPathConditions, `multiple "Denied" conditions present`),
+			},
+		},
```

Comment:
Aren't these tests duplicating the tests for `ValidateCertificateRequestCondition` above? 
If so, I'd be inclined to remove them.

Reply:
Kind of. They _are_ testing the same thing, though one is for CREATE func and the other UPDATE func. Keen to test the same behaviour where they are the same by duplicating some tests.